crates/bevy_ecs/src/reflect.rs

```diff
@@ -418,15 +418,30 @@ impl ReflectMapEntities {
         world: &mut World,
         entity_map: &EntityMap,
     ) -> Result<(), MapEntitiesError> {
-        (self.map_entities)(world, entity_map)
+        (self.map_entities)(world, entity_map, entity_map.values().collect())
+    }
+
+    /// This is like `map_entities`, but only applied to specific entities, not all values in the `EntityMap`.
+    ///
+    /// This is useful for when you only want to apply the `MapEntity` logic to specific entities. For example,
+    /// a scene can be loaded with `Parent` components, but then a `Parent` component can be added to these entities
+    /// after they have been loaded. If you reload the scene and used plain `map_entities`, those `Parent` components
+    /// with already valid entity references could be updated to point at something else entirely.
```

Comment:
At first I left this here as a habit from work where if you have a public API that might be consumed by others, leave it alone. But I realize that's not TheBevyWay(tm) right now, so we can go ahead and replace it.

So I used `ripgrep` to find invocations, and I found out the only other place is in `Scene::write_to_world_with` (Not `DynamicScene::write_to_world_with`). This bug does not exist in this case because `EntityMap` is not passed to this code; it seems you can't do a "reload" with a `Scene`. I'll modify the code there to pass in the entity values though to use this interface.